src/test/java/org/synyx/urlaubsverwaltung/overtime/web/OvertimeControllerTest.java

```diff
@@ -55,6 +59,46 @@ public void setUp() {
         sut = new OvertimeController(overtimeService, personService, validator, departmentService);
     }
 
+    @Test
+    public void postRecordOvertimeShowsFormIfValidationFails() throws Exception {
+
+        final Person signedInPerson = new Person();
+        signedInPerson.setPermissions(Collections.singletonList(OFFICE));
+
+        when(personService.getSignedInUser()).thenReturn(signedInPerson);
+
+        doAnswer(invocation -> {
+            Errors errors = invocation.getArgument(1);
+            errors.rejectValue("person", "errors");
+            return null;
+        }).when(validator).validate(any(), any());
+
+        perform(post("/web/overtime"))
+            .andExpect(model().attribute("overtime", instanceOf(OvertimeForm.class)))
+            .andExpect(view().name("overtime/overtime_form"));
```

Comment:
imho brauchts hier noch noch eine prüfung, dass `validator.validate` auch aufgerufen wurde. weil die implementierung kann ich dahingegen ändern, dass `andExpect` erfolgreich bleibt, aber nie `validator.validate` aufgerufen wird.

also wenn ich `validator.validate` lösche in der implementierung, schlägt der test fehl weil `final Person person = overtimeForm.getPerson();` `null` ist und `Overtime` eine person benötigt.
der test hier dreht sich aber explizit um die validation.

Reply:
Danke fuer deine Meinung :) Ich habe es verstanden und schon korrigiert.